src/shared_realm.hpp

```diff
@@ -84,6 +84,17 @@ enum class SchemaMode : uint8_t {
     // are allowed to be missing from the file.
     ReadOnly,
 
+    // Open the Realm in read-only mode, transactions are not allowed to
+    // be performed on the Realm instance. The schema of the existing Realm
+    // file won't be changed through this Realm instance. Extra tables and
+    // extra properties are allowed in the existing Realm schema. The
+    // difference of indexes is allowed as well. Other schema differences
+    // than those will cause an exception. This is different from ReadOnly
+    // mode, sync Realm can be opened with ReadOnlyAlternative mode. Changes
+    // can be made to the Realm file through another writable Realm instance.
+    // Thus, notifications are also allowed in this mode.
+    ReadOnlyAlternative,
```

Comment:
I spent an hour to come with this name ... actually i feel the original `ReadOnly` is more strict than this one since it doesn't even allow `advance_read`. I am open for other better names :P